ci/build_windows.py

```diff
@@ -0,0 +1,253 @@
+        libs = list(glob.iglob('**/*.lib', recursive=True))
+        dlls = list(glob.iglob('**/*.dll', recursive=True))
+        os.makedirs(pkgdir_lib, exist_ok=True)
+        for lib in libs:
+            logging.info("packing lib: %s", lib)
+            shutil.copy(lib, pkgdir_lib)
+        for dll in dlls:
+            logging.info("packing dll: %s", dll)
+            shutil.copy(dll, pkgdir_lib)
+        os.chdir(get_mxnet_root())
+        logging.info('packing python bindings')
+        copy_tree('python', j(pkgdir, 'python'))
+        logging.info('packing headers')
+        copy_tree('include', j(pkgdir, 'include'))
+        copy_tree(j('3rdparty','dmlc-core','include'), j(pkgdir, 'include'))
+        copy_tree(j('3rdparty','mshadow', 'mshadow'), j(pkgdir, 'include', 'mshadow'))
+        copy_tree(j('3rdparty','tvm','nnvm', 'include'), j(pkgdir,'include', 'nnvm', 'include'))
+        logging.info("Compressing package: %s", pkgfile)
+        check_call(['7z', 'a', pkgfile, pkgdir])
+
+
+def nix_build(args):
+    path = args.output
+    os.makedirs(path, exist_ok=True)
+    with remember_cwd():
+        os.chdir(path)
+        logging.info("Generating project with CMake")
+        check_call("cmake \
+            -DUSE_CUDA=OFF \
+            -DUSE_OPENCV=OFF \
+            -DUSE_OPENMP=OFF \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -GNinja ..", shell=True)
+        check_call("ninja", shell=True)
+
+def main():
+    logging.getLogger().setLevel(logging.INFO)
+    logging.basicConfig(format='%(asctime)-15s %(message)s')
+    logging.info("MXNet Windows build helper")
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-o", "--output",
+        help="output directory",
+        default='build',
+        type=str)
+
+    parser.add_argument("--vcvars",
+        help="vcvars batch file location, typically inside vs studio install dir",
+        default=r'c:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsx86_amd64.bat',
+        type=str)
+
+    parser.add_argument("--arch",
+        help="architecture",
+        default='x64',
+        type=str)
+
+    parser.add_argument("-f", "--flavour",
+        help="build flavour",
+        default='WIN_CPU',
+        choices=[x.name for x in BuildFlavour],
+        type=str)
+
+    args = parser.parse_args()
+    logging.info("Build flavour: %s", args.flavour)
+
+    system = platform.system()
+    if system == 'Windows':
+        logging.info("Detected Windows platform")
+        if 'OpenBLAS_HOME' not in os.environ:
+            os.environ["OpenBLAS_HOME"] = "C:\\mxnet\\openblas"
+        if 'OpenCV_DIR' not in os.environ:
+            os.environ["OpenCV_DIR"] = "C:\\mxnet\\opencv_vc14"
+        if 'CUDA_PATH' not in os.environ:
+            os.environ["CUDA_PATH"] = "C:\\CUDA\\v8.0"
+        windows_build(args)
+
+    elif system == 'Linux' or system == 'Darwin':
```

Comment:
Ah, now I get it. But why are we offering unix compilation in a file which is only for windows compilation? Seems like we're mixing concerns here a bit.

Reply:
this is not exposed to users right now.